### PR TITLE
feat(ci): Collapse long logs in section

### PR DESCRIPTION
**Pull request by chouetz**
### What does this PR do?
Create a context manager to wrap logs in gitlab

### Motivation
Improve readability of the logs

### CODE REVIEW

**Review by jszwedko — APPROVED** [review]

File: tasks/libs/common/utils.py [hunk 1]
```diff
@@ -700,3 +700,13 @@ def is_pr_context(branch, pr_id, test_name):
...
         return False
     return True
+
+
+@contextmanager
+def collapsed_section(section_name):
+    section_id = section_name.replace(" ", "_")
+    try:
+        print(f"\033[0Ksection_start:{int(time.time())}:{section_id}[collapsed=true]\r\033[0K{section_name + '...'}")
+        yield section_id
```
Comment (FlorentClarret) [inline tasks/libs/common/utils.py:710]: I don't mind yielding it, but what could the user do with this id?

File: tasks/libs/common/utils.py [hunk 2]
```diff
@@ -700,3 +700,13 @@ def is_pr_context(branch, pr_id, test_name):
         print(f"PR not found, skipping check for {test_name}.")
         return False
     return True
+
+
+@contextmanager
+def collapsed_section(section_name):
```
Comment (KSerrania) [inline tasks/libs/common/utils.py:706]: Should this detect if we're running in CI? I assume printing these section headers and footers locally will have no effect, and will only add noise to the logs.

**Review by vickenty — APPROVED** [review]

File: tasks/libs/common/utils.py (hunk 1 above)
Comment (chouetz) [reply tasks/libs/common/utils.py:710]: Indeed it was just to follow the `context manager` pattern, this is probably useless

File: tasks/libs/common/utils.py (hunk 2 above)
Comment (chouetz) [reply tasks/libs/common/utils.py:706]: Correct I'll fix this

**Review by FlorentClarret — APPROVED** [review]

**Review by spencergilbert — APPROVED** [review]

**Review by clarkb7 — APPROVED** [review]
neat!
I don't see the `Running unit tests` section in the tests_windows-x64 job. Does it need a env var passed through?

**Comment by chouetz** [comment]
> neat! I don't see the `Running unit tests` section in the tests_windows-x64 job. Does it need a env var passed through?

Indeed you are right, we are missing env variable to make it understand we are in the gitlab-ci context, fixing this

**Pull request merged**